### PR TITLE
[Distributed] fix deinitialization of result buffer

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please build toolchain macOS

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 1]
```diff
@@ -555,9 +555,12 @@ extension DistributedActorSystem {
         errorCode: .typeDeserializationFailure)
     }
 
+    var executeDistributedTargetHasThrown = false
     func doDestroyReturnTypeBuffer<R>(_: R.Type) {
       let buf = resultBuffer.assumingMemoryBound(to: R.self)
-      buf.deinitialize(count: 1)
+      if R.self != Void.self && !executeDistributedTargetHasThrown {
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:561]: I actually wonder whether this is reliable, technically we could put result into the buffer and throw afterwards?

Comment (ktoso) [reply stdlib/public/Distributed/DistributedActorSystem.swift:561]: Hmmm, would we...? Maybe... I'll be digging deeper into this today, could not reproduce the issue so far. But definitely seems like on some platforms may be the not initialzied memory then getting deinitialized makes us crash, so there has to be a way for us to know if to deinit or not hmm

Comment (xedin) [reply stdlib/public/Distributed/DistributedActorSystem.swift:561]: One idea I had is to set a bit on the buffer pointer itself to indicate that was initialized, that way even if something throws we'd still de-initialize the memory.

**Comment by ktoso** [comment]
I actually managed to reproduce this on Linux finally and yeah this seems good enough to fix the issue so I'll go ahead landing this so we can get it in quickly.

**Comment by ktoso** [comment]
@swift-ci please test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
Removed the Void check, it didn't matter it seems

**Pull request merged**